### ISSUE TITLE
Can't upload more than 6 images

### ISSUE THREAD

**Issue by Firesphere** [colymba/GridFieldBulkEditingTools#117]
Anything more just simply hangs and isn't finished or linked.

**Comment by colymba** [comment]
Weird. Is there an error, a log, anything? What did the request returns?

**Comment by Firesphere** [comment]
Nope, nothing, no error, no log, just.... nothing, request doesn't finish, even though my upload-limit is 100MB.
Uploading via Assets works perfectly fine.

**Comment by colymba** [comment]
Can we have a look at your code. And which browser is this with?

**Comment by Firesphere** [comment]
Browsers:
Chrome, Edge and Firefox.
Code GalleryPage:
```

	private static $has_one = array(
		'CoverImage' => 'Image',
	);

	private static $has_many = array(
		'GalleryImages' => 'GalleryImage'
	);

	public function getCMSFields()
	{

		$fields = parent::getCMSFields();

		/** @var GridFieldConfig_RecordEditor $gridFieldConfig */
		$gridFieldConfig = GridFieldConfig_RecordEditor::create();
		$gridFieldConfig->addComponent(new GridFieldBulkUpload());
		$gridFieldConfig->addComponent(new GridFieldOrderableRows('SortOrder'));
		$gridFieldConfig->addComponent(new GridFieldEditableColumns());

		$gridfield = new GridField("GalleryImages", "Gallery Images", $this->GalleryImages()->sort("SortOrder"), $gridFieldConfig);

		$fields->addFieldToTab('Root.GalleryImages', $gridfield);

		return $fields;

	}
```
Code GalleryImage:
```
	private static $db = array(
		'SortOrder' => 'Int',
		'Title'     => 'Varchar'
	);

	// One-to-one relationship with gallery page
	private static $has_one = array(
		'Image'       => 'Image',
		'GalleryPage' => 'GalleryPage'
	);

	private static $sort_order = 'SortOrder, Created';

	// tidy up the CMS by not showing these fields
	public function getCMSFields()
	{
		$fields = parent::getCMSFields();
		$fields->removeFieldFromTab("Root.Main", "GalleryPageID");
		$fields->removeFieldFromTab("Root.Main", "SortOrder");

		return $fields;
	}

	// Tell the datagrid what fields to show in the table
	private static $summary_fields = array(
		'Title'     => 'Title',
		'Thumbnail' => 'Thumbnail'
	);

	// this function creates the thumnail for the summary fields to use
	public function getThumbnail()
	{
		return $this->Image()->CMSThumbnail();
	}

	public function getThumb()
	{
		return $this->Image()->FillMax(300, 300);
	}

	public function onAfterWrite()
	{
		parent::onAfterWrite();
		$this->resizeImages();
	}

	/**
	 * If there's a max-size set in the SiteConfig ({width}x{height}) for the image, resize the image.
	 * This saves space on the hosting and prevents huge high-res images to stay online for no reason.
	 */
	public function resizeImages()
	{
		$size = array(1024, 768);
		if ($this->Image()->getWidth() > $size[0]) {
			$maxSized = $this->Image()->ScaleWidth($size[0]);

			unlink(Director::baseFolder() . '/' . $this->Image()->getFilename());
			copy(Director::baseFolder() . '/' . $maxSized->getFilename(), Director::baseFolder() . '/' . $this->Image()->getFilename());
			unlink(Director::baseFolder() . '/' . $maxSized->getFilename());
		}
		if ($this->Image()->getHeight() > $size[1]) {
			$maxSized = $this->Image()->ScaleHeight($size[1]);

			unlink(Director::baseFolder() . '/' . $this->Image()->getFilename());
			copy(Director::baseFolder() . '/' . $maxSized->getFilename(), Director::baseFolder() . '/' . $this->Image()->getFilename());
			unlink(Director::baseFolder() . '/' . $maxSized->getFilename());
		}
	}
```

**Comment by colymba** [comment]
Which version of the module, cms and framework address you using?

**Comment by Firesphere** [comment]
3.2.1, dev-master.

**Comment by purplespider** [comment]
@Firesphere Did you ever get to the bottom of this?

A client of mine is experiencing a similar issue, it uploads 11 images and then hangs. Works fine when I try it with the same images.

The only error I can find relating it is a 404 from modsecurity:
`[modsecurity] [client 90.254.145.180] [domain www.connellmcfadden.com] [404] [/nobody/20161006/20161006-1119/20161006-111925-V-YlLLJPrF8ACY9yflMAAAAO] (null)`

**Comment by colymba** [comment]
just out of curiosity, have you tried in both _parallel_ and _sequential_ upload mode?

**Comment by purplespider** [comment]
I always have it in sequential mode so that the images are kept in the the expected order, but I've not tried it in parallel mode.

I don't understand why I can't reproduce it myself, though. My client gets it every time after the 11th image and even when I use the same browser and same image files, it works fine for me.

**Comment by Firesphere** [comment]
I don't have ModSecurity, as I'm on Nginx systems. Problem persists up to 3.4.1 so far.

I'll have a look if I can find anything next time I have to upload a bunch of pictures.

I do not encounter this issue on my local machine, so I think it might be something in the PHP config that's bombing.

**Comment by colymba** [comment]
It does sound to me that's it's a server config issue.
If you manage to find a log or something about the error we can continue investigating.

**Comment by colymba** [comment]
I know it's quite an old issue, but from my test on the latest version this is not an issue anymore.. probably because of all the major rewrite.... feel free to reopen or create a new issue if something pops up.
